app/src/lib/menu-update.ts

```diff
@@ -248,7 +248,7 @@ function getRepositoryMenuBuilder(state: IAppState): MenuStateBuilder {
     )
     menuStateBuilder.setEnabled(
       'create-branch',
-      !tipStateIsUnknown && !branchIsUnborn
+      !tipStateIsUnknown && !branchIsUnborn && !onDetachedHead
```

Comment:
I'm not sure this is the complete fix because creating a branch from a detached HEAD is still a valid scenario. The issue description in https://github.com/desktop/desktop/issues/6335 shows the steps a user might do (requires terminal usage).